### PR TITLE
PoC for the Community Organization v2

### PR DESCRIPTION
**Pull request by pjedrzejewski**
Hi!

Our community is growing, Sylius is becoming more and more mature and obviously, there is an infinite amount of work to do. In order to get the community more involved and better organized, I propose to adopt the following rules, processes and release cycle. This is only a PoC, but if you will decide it makes sense - we will use it as a base and let it evolve, as usually.

Most important things included in this PR:

* New Sylius Core Team - I defined new roles and assigned them to familiar faces in the community. We want to include more people and definitely more from outside of the commercial company. Stay tuned! We first want to define some strategy for the community organization and then we can grow it.

* Adopting the new Release Process - While the final date for 1.0.0 is not set yet (because it also depends on the release process we adopt), I'd like to already define a time based release cycle - inspired by the Symfony project. Everything is described in the docs, so let me know what do you think. The main goal is to make the development more predictable for both the developers and business. At the same time - allow us to improve the platform and introduce new features.

* Efforts - Sometimes things are not simple enough for a pair of issue and PR. :) We have used this kind of approach in the past and it worked well. An example effort is here: #8106.

Let me know what do you think! I hope to iterate on this based on your feedback and internal discussions of the Core Team. After we have the MVP, we release a blog post and kickstart more Efforts.

### CODE REVIEW

**Comment by kochen** [comment]
Great initiative and good timing (being closer to a stable release)!

I think it would be a good idea to think about a community forum, where
members could get and provide support to and from each other.

Mati

File: docs/contributing/install.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    secret (EDITME):
+    locale (en_US):
+    currency (USD):
+
+After everything is in place, run the following commands:
+
+.. code-block:: bash
+
+    $ cd sylius # Move to the newly created directory
+    $ php bin/console sylius:install
```
Comment (pamil) [inline docs/contributing/install.rst]: We can omit `php` part of the command.

File: docs/contributing/organization/core-team.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+members must wait at least 6 months before re-applying.
+
+Sylius Core Team Rules and Protocol Amendments
+----------------------------------------------
+
+The rules described in this document may be amended at anytime by the **Project Lead**.
+
+.. _`pjedrzejewski`: https://github.com/pjedrzejewski/
+.. _`pamil`: https://github.com/pamil/
+.. _`CoderMaggie`: https://github.com/CoderMaggi/
```
Comment (pamil) [inline docs/contributing/organization/core-team.rst]: Isn't there `e` missing at the end of the URL?

File: docs/contributing/organization/release-process.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Standard Versions
+~~~~~~~~~~~~~~~~~
+
+A **Standard Minor Version** is maintained for an *eight month* period for bug
+fixes, and for a *fourteen month* period for security issue fixes.
+
+Long Term Support Versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Every two years, a new **Long Term Support Version** (usually abbreviated as "LTS")
```
Comment (pamil) [inline docs/contributing/organization/release-process.rst]: Shall it be the last minor version of a given major one?

File: docs/contributing/organization/release-process.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+is published. Each LTS version is supported for a *three year* period for bug
+fixes, and for a *four year* period for security issue fixes.
+
+Backward Compatibility
+----------------------
+
+Our Backward Compatibility Promise is very
+strict and allows developers to upgrade with confidence from one minor version
```
Comment (pamil) [inline docs/contributing/organization/release-process.rst]: We should specify what exactly do we guarantee, something like http://symfony.com/doc/current/contributing/code/bc.html

File: docs/contributing/organization/vision-and-strategy.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+* **New Feature** - PRs, which implement new features;
+* **RFC** - Discussions about potential changes or new features;
+* **Documentation Request** - New documentation request;
+* **BC Break** - Patches, which break backwards compatibility;
+* **Minor** - Typo fixes, spelling, grammar, commas and other valuable, although not crucial changes;
+* **Maintenance** - Travis configurations, READMEs, releases, etc.;
+* **Optimization** - Patches which optimize the performance of the platform;
+* **Environment** - Environment (OS, databases, libraries, etc.) specific issues;
+* **UI** - User Interface related issues;
+* **Translations** - Everything related to transalting the UI;
```
Comment (pamil) [inline docs/contributing/organization/vision-and-strategy.rst]: `transalting` -> `translating`

File: docs/contributing/organization/vision-and-strategy.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+5. PR contains appropriate UPGRADE file updates if necessary;
+6. PR is properly labeled and milestone is assigned if needed;
+7. All required checks are passing. It is green!
+
+Certain PRs can only be merged by the Project Lead:
+
+* BC Breaks;
+* Introducing new components, bundles or high level architecture layers;
+* Renaming existing components;
+* If in doubt, ask your friendly neighborhood Project Lead.
```
Comment (pamil) [inline docs/contributing/organization/vision-and-strategy.rst]: `neighborhood`? 💃

Comment (pjedrzejewski) [inline docs/contributing/organization/vision-and-strategy.rst]: Man, it's a low-quality pun. Friendly Neighbourhood Spider-Man. XDDD

File: docs/contributing/organization/release-process.rst (hunk 3 above)
Comment (mbabker) [inline docs/contributing/organization/release-process.rst]: That should probably be explained a bit, yes.

**Comment by mbabker** [comment]
So one thought based on a first read over this, and this is coming with a few years of having been the release lead and on the "core team" for Joomla.

At one point our release cycle was pretty similar to what is proposed here (new minor version every six months, new major series every two years) and toward the end of us following that cycle it really started to feel like we were just bumping version numbers for the sake of bumping version numbers, especially as it related to starting a new major series.  So, depending on whether the plan is to distribute Sylius as an application or a framework, I'm not so sure following this aspect of Symfony's release plan works as efficiently if you're doing the application based approach.  I think the new minor release every six months plan can work when managed well (so far it seems fine with Drupal), but for an application based release, I would re-evaluate this part a bit.

File: docs/contributing/organization/release-process.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+of Sylius to the next one.
+
+Whenever keeping backward compatibility is not possible, the feature, the
```
Comment (mbabker) [inline docs/contributing/organization/release-process.rst]: Perhaps adding a caveat here for security related issues?  Symfony 3.3 is including a B/C break based on https://github.com/symfony/symfony/issues/20178.  Obviously even in that case the break should be kept to the absolute minimum essential change, but I do think patching a security issue is a valid B/C break when absolutely necessary.

File: docs/contributing/organization/core-team.rst [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+All the Sylius Core Team members are long-time contributors with solid technical
+expertise and they have demonstrated a strong commitment to drive the project
+forward.
+
+This document states the rules that govern the Sylius Core Team. These rules
+are effective upon publication of this document and all Sylius Core Team members
+must adhere to said rules and protocol.
+
+Core Team Organization
```
Comment (stefandoorn) [inline docs/contributing/organization/core-team.rst]: Would it be an idea to assign someone with 'Plugin Lead' focussing mainly on the proper plugin points inside Sylius to allow for swift development of plugins. Next to this, maybe also someone / several people should be in contact with certain plugin developer to help them improve important plugins?

File: docs/contributing/organization/release-process.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+code & documentation hosted on the main ``Sylius/Sylius`` `Git repository`_).
+
+Sylius manages its releases through a *time-based model* and follows the
+`Semantic Versioning`_ strategy:
+
+* A new Sylius minor version (e.g. 1.1, 2.4, 3.2) comes out every *six months*:
+* A new Sylius major version (e.g., 2.0, 3.0) comes out every *two years* and
+  it's released at the same time of the last minor version of the previous major
+  version.
+
```
Comment (stefandoorn) [inline docs/contributing/organization/release-process.rst]: Can also be something said about patches? I'm now following dev-master in my current project, which I update every 2-3 weeks. I would like to stay up to date with the project on stable patches with easy upgrades. Upgrading every 6 months once, means an huge changelog and potentially lots of updates to your own customisations. The less big the updates are or the more options you have to keep up to date, the better in my opinion. Nearly no project has every customisation covered with behat scenarios, unit / integration tests, so you need to read all the changes anyway, so if that's a small diff it's relatively easy.

File: docs/contributing/organization/vision-and-strategy.rst (hunk 6 above)
Comment (stefandoorn) [inline docs/contributing/organization/vision-and-strategy.rst]: License check, there are tools like this, could be helpful: https://clabot.github.io/

File: docs/contributing/organization/release-process.rst (hunk 9 above)
Comment (pamil) [inline docs/contributing/organization/release-process.rst]: Shall we consider releasing patches every two weeks?

**Comment by pjedrzejewski** [comment]
Thanks for the feedback! I will make appropriate changes and we shall iterate on this once more. :)

File: docs/contributing/organization/vision-and-strategy.rst [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+Every Effort should have a main Effort Issue opened on GitHub, which contains all details about particular task. It should also have an Effort Lead(s) assigned.
+
+Individual tasks that push the effort forward shall be created as separate issues, but always referenced in the Effort Issue.
+
+An example of Efforts: "Optimization of the Checkout Process", "Integrate All The Things" (better foundation for integrations) or whatever creative name you can come up with.
+
+Anyone can kickstart a new Effort, but first one should discuss it with the Core Team Members (on Slack) to get proper guidance.
+
+Labels (Issue Types)
```
Comment (gabiudrescu) [inline docs/contributing/organization/vision-and-strategy.rst]: who is in charge of putting labels on issues?

File: docs/contributing/organization/infrastructure.rst [hunk 11]
```diff
@@ -0,0 +1,31 @@
+Infrastructure
+==============
+
+In order to better organize the community and automate some tasks, we the following services and applications.
```
Comment (adamelso) [inline docs/contributing/organization/infrastructure.rst]:
```diff
- we the following services and applications
+ we use the following services and applications
```

**Comment by pamil** [comment]
Solved by #8951.

**Pull request closed without merging**